pulsar-common/src/main/java/com/yahoo/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -528,4 +529,23 @@ public int compareTo(LongPair o) {
             }
         }
     }
+    
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append('{');
+        final AtomicBoolean first = new AtomicBoolean(true);
```

Comment:
Do we need to use an atomicBoolean here? Can't we use something like,

String s = "";
foreach() {
sb.append(s);
s = ", ";
}

I'm hoping compiler will optimize the reinitialization of s.